implementations/micrometer-registry-datadog/src/main/java/io/micrometer/datadog/DatadogMeterRegistry.java

```diff
@@ -78,17 +82,63 @@ public DatadogMeterRegistry(DatadogConfig config, Clock clock) {
      */
     @Deprecated
     public DatadogMeterRegistry(DatadogConfig config, Clock clock, ThreadFactory threadFactory) {
-        this(config, clock, threadFactory, new HttpUrlConnectionSender(config.connectTimeout(), config.readTimeout()));
+        this(config, clock, threadFactory, new HttpUrlConnectionSender(config.connectTimeout(), config.readTimeout()),
+                null);
+    }
+
+    /**
+     * @param config Configuration options for the registry that are describable as
+     * properties.
+     * @param clock The clock to use for timings.
+     * @param threadFactory The thread factory to use to create the publishing thread.
+     * @deprecated Use {@link #builder(DatadogConfig)} instead.
+     */
+    @Deprecated
+    public DatadogMeterRegistry(DatadogConfig config, Clock clock, ThreadFactory threadFactory,
+            @Nullable StatsDClient customClient) {
+        this(config, clock, threadFactory, new HttpUrlConnectionSender(config.connectTimeout(), config.readTimeout()),
+                customClient);
     }
 
-    private DatadogMeterRegistry(DatadogConfig config, Clock clock, ThreadFactory threadFactory,
-            HttpSender httpClient) {
+    /**
+     * Truthy if the uri is defined for a dogstatsd reporting configuration.
+     * @return Returns true if this is a non https scheme target, to indicate to talk via
+     * dogstatsd.
+     */
+    static boolean isStatsd(String uri) {
+        String scheme = URI.create(uri).getScheme();
```

Comment:
Are we ok if the uri creation throws an exception (when the uri is invalid) ? I would presume that that's perfectly fine

Reply:
yeah, this was intentional, a bad URI could potentially drop data. The same is true of http(s) if the wrong datacenter or path is used, this would provide faster feedback if something really bad happened